### PR TITLE
README and dev doc updates

### PR DESCRIPTION
**Pull request by josh-berry**
- Consolidate all dev docs into CONTRIBUTING.md
- Document the release process (such as I understand it)

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -28,6 +28,15 @@ Build the `temporal` binary:
 go build ./cmd/temporal
 ```
 
+## Generate docs
+
+```bash
+go build -o temporal-doc-gen ./cmd/docgen
```
Comment (bergundy) [inline CONTRIBUTING.md:34]: I think there's a target in the Makefile (which I'm totally fine with removing BTW).

Another option is running with `go run`.

```
go run ./cmd/docgen
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -28,6 +28,15 @@ Build the `temporal` binary:
...
+./temporal-doc-gen
+```
+
+Docs are generated as Markdown files in `./docs`.
```
Comment (bergundy) [inline CONTRIBUTING.md:38]: We should say what to do with these generated docs, how do they get used in our documentation site?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -73,3 +82,22 @@ To verify that this file is up to date execute:
 ```bash
 go run ./cmd/licensecheck
 ```
+
+## Release process
+
+If you're a Temporal engineer / code owner, here's how to do a release:
+
+1. Do some manual testing to make sure things look good--start a dev-server, run
+   some workflows, etc.
```
Comment (bergundy) [inline CONTRIBUTING.md:91]: Shouldn't the automated tests cover this?
I would rather we replace this with "verify that CI passes for the commit hash".

Comment (bergundy) [reply CONTRIBUTING.md:91]: (Which I see that you have below).

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -73,3 +82,22 @@ To verify that this file is up to date execute:
...
+2. Make sure CI is passing on `main`.
+3. Create a tag of the form `vX.Y.Z` off of `main` and push it to GitHub.
+4. GoReleaser will pick up the tag and build the release artifacts.
+5. Open a PR against Homebrew to modify [the temporal formula] to fetch the
+   source tarball using the latest release from GitHub.  Don't forget to update
+   the checksum of the source tarball. ([Example PR])
+   - Homebrew should take care of rebottling the formula and updating the bottle
+     checksums; you don't need to do this yourself.
+6. TODO: Describe how to update `temporal.download` links
```
Comment (bergundy) [inline CONTRIBUTING.md:100]: There's nothing to update there.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -73,3 +82,22 @@ To verify that this file is up to date execute:
...
+
+[the temporal formula]: https://github.com/Homebrew/homebrew-core/blob/master/Formula/t/temporal.rb
```
Comment (bergundy) [inline CONTRIBUTING.md:102]: May want to use a permalink here in case this changes for whatever reason.

File: CONTRIBUTING.md (hunk 2 above)
Comment (josh-berry) [reply CONTRIBUTING.md:38]: Good point. I don't know the answer to this, I'll do some research.

File: CONTRIBUTING.md (hunk 3 above)
Comment (josh-berry) [reply CONTRIBUTING.md:91]: Yeah, the problem is nobody seems to fully-trust that CI is "good enough" right now—Chad's told me multiple times we need to do manual testing on things like server updates.

Comment (bergundy) [reply CONTRIBUTING.md:97]: It's fine as long as there's a step to verify that CI is green.

File: CONTRIBUTING.md (hunk 1 above)
Comment (bergundy) [reply CONTRIBUTING.md:40]: I'd switch to `go run` though.

**Review by bergundy — APPROVED** [review]
Just one nit, otherwise LGTM.
Thanks.

**Comment by josh-berry** [comment]
> Just one nit, otherwise LGTM. Thanks.

Just to clarify—is the nit your comment above about how to publish docs, or was there another one?

File: CONTRIBUTING.md (hunk 2 above)
Comment (josh-berry) [reply CONTRIBUTING.md:43]: Added another item for this to the release process.

**Pull request merged**